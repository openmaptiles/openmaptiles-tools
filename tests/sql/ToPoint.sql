-- ToPoint
SELECT ToPoint(ST_GeomFromText('POINT(0 0)',900913)) AS v1;	-- 010100002031BF0D0000000000000000000000000000000000
SELECT ToPoint(ST_GeomFromText('POLYGON EMPTY',900913)) AS null_v2;	-- \\N
SELECT ToPoint(ST_GeomFromText('POLYGON((0 0, 10 0, 10 10, 0 10, 0 0))',900913)) AS v3;	-- 010100002031BF0D0000000000000014400000000000001440
SELECT ToPoint(ST_GeomFromText('POLYGON((0 0, 10 0, 0 10, 10 10, 0 0))',900913)) AS v4;	-- 010100002031BF0D0000000000000014400000000000000440
SELECT ToPoint(ST_GeomFromText('MULTIPOLYGON(((0 0, 10 0, 10 10, 0 10, 0 0)))',900913)) AS v5;	-- 010100002031BF0D0000000000000014400000000000001440
SELECT ToPoint(ST_GeomFromText('MULTIPOLYGON(((0 0, 10 0, 10 10, 0 10, 0 0)), ((20 20, 30 20, 30 30, 20 30, 20 20)))',900913)) AS v6;	-- 010100002031BF0D0000000000000014400000000000001440
SELECT ST_AsText(ToPoint(ST_GeomFromText('POLYGON((50 5,10 8,10 10,100 190,150 30,150 10,50 5))',900913))) AS v7;	-- POINT(92.5 110)
