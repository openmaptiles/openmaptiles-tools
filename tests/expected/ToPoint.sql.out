                         v1                         
----------------------------------------------------
 010100002031BF0D0000000000000000000000000000000000
(1 row)

 null_v2 
---------
 
(1 row)

                         v3                         
----------------------------------------------------
 010100002031BF0D0000000000000014400000000000001440
(1 row)

                         v4                         
----------------------------------------------------
 010100002031BF0D0000000000000014400000000000000440
(1 row)

                         v5                         
----------------------------------------------------
 010100002031BF0D0000000000000014400000000000001440
(1 row)

                         v6                         
----------------------------------------------------
 010100002031BF0D0000000000000014400000000000001440
(1 row)

       v7        
-----------------
 POINT(92.5 110)
(1 row)

